skimage/morphology/footprints.py

```diff
@@ -959,3 +959,69 @@ def star(a, dtype=np.uint8):
     footprint[footprint > 0] = 1
 
     return footprint.astype(dtype)
+
+
+def mirror_footprint(footprint):
+    """Mirror each dimension in the footprint.
+
+    Parameters
+    ----------
+    footprint : ndarray or tuple
+        The input footprint or sequence of footprints
+
+    Returns
+    -------
+    inverted : ndarray or tuple
+        The footprint, mirrored along each dimension.
+
+    Examples
+    --------
+    >>> footprint = np.array([[0, 0, 0],
+    ...                       [0, 1, 1],
+    ...                       [0, 1, 1]], np.uint8)
+    >>> mirror_footprint(footprint)
+    array([[1, 1, 0],
+           [1, 1, 0],
+           [0, 0, 0]], dtype=uint8)
+
+    """
+    if _footprint_is_sequence(footprint):
+        return tuple((mirror_footprint(fp), n) for fp, n in footprint)
+    footprint = np.asarray(footprint)
+    return footprint[(slice(None, None, -1),) * footprint.ndim]
+
+
+def pad_footprint(footprint, *, pad_end=True):
```

Comment:
I refactored the parameter `right` to `pad_end` and also made it keyword only. Basically see the description in https://github.com/scikit-image/scikit-image/pull/6695/commits/1bc9a7ea2588472475cd2a937996caf7d737cdb5 for the reasoning. Just highlighting this because it doesn't hurt to deliberate a bit about new API we are introducing.

Reply:
Yes, this is much better!